src/main/java/robot/subsystems/turret/Turret.java

```diff
@@ -0,0 +1,149 @@
+ */
+public class Turret extends Subsystem {
+    private TalonSRX master = new TalonSRX(MASTER);
+    private double angle;
+    private Direction direction;
+
+    /**
+     * configures the encoder and PID constants.
+     */
+    public Turret() {
+        direction = Direction.RIGHT;
+        master.configSelectedFeedbackSensor(FeedbackDevice.CTRE_MagEncoder_Relative, 0, TIMEOUT_MS);
+        master.config_kP(TALON_PID_SLOT, KP, TIMEOUT_MS);
+        master.config_kI(TALON_PID_SLOT, KI, TIMEOUT_MS);
+        master.config_kD(TALON_PID_SLOT, KD, TIMEOUT_MS);
+        master.config_kF(TALON_PID_SLOT, KF, TIMEOUT_MS);
+        master.setInverted(IS_MASTER_INVERTED);
+        master.configPeakCurrentLimit(MAX_CURRENT);
+        master.configForwardLimitSwitchSource(LimitSwitchSource.FeedbackConnector, LimitSwitchNormal.NormallyOpen);
+        reset();
+        master.setSelectedSensorPosition(0);
+    }
+
+    @Override
+    public void initDefaultCommand() {
+        setDefaultCommand(new TurnTurret(45));
+    }
+
+    /**
+     * set the direction of the turret,
+     * either left or right.
+     *
+     * @param direction the direction that the turret will move.
+     */
+    public void setDirection(Direction direction) {
+        this.direction = direction;
+    }
+
+    /**
+     * get the current angle from the controller
+     *
+     * @return the angle of the turret
+     */
+    public double getAngle() {
+        return convertTicksToDegrees(master.getSelectedSensorPosition());
+    }
+
+    /**
+     * change the angle to the desired angle,
+     * if you would like to use the same Direction.
+     * the value can be between -180 to 180 degrees.
+     *
+     * @param targetAngle the desired angle.
+     */
+    public void setTargetAngle(double targetAngle) {
+        if (Double.compare(targetAngle, angle) == 0) return;
+        while (targetAngle > 180)
+            targetAngle -= 360;
+        while (targetAngle < -180) {
+            targetAngle += 360;
+        }
+        //TODO: check which side is the positive one
+        this.angle = targetAngle;
+        changeAngle(angle);
+    }
+
+    /**
+     * applying power to the controller for moving the turret.
+     *
+     * @param angle the desired angle
+     */
+    private void changeAngle(double angle) {
+        master.set(ControlMode.MotionMagic, convertDegreesToTicks(angle));
+    }
+
+    /**
+     * @return return if the state of the the Hall Effect sensor is Closed.
+     */
+    private boolean getHallEffect() {
+        return master.getSensorCollection().isRevLimitSwitchClosed();
+    }
+
+    /**
+     * set encoder position to the Hall Effect position.
+     */
+    public void adjustEncoderPosition() {
+        if (getHallEffect()) {
+            master.setSelectedSensorPosition((int) convertDegreesToTicks(HALL_EFFECT_POSITION), 0, TIMEOUT_MS);
+        }
+    }
+
+    /**
+     * reset the turret so it will be at the same position every run.

```

Comment:
```suggestion
     * Zero the turret.
```
Javadoc is not the place for explanation _why_ a code exists. You can write it like that:
```
/**
* short explanation about what the method does
* [empty line]
* longer explanation about everything eles
*/
```